src/main/proto/tech/figure/loan/v1beta1/loan.proto

```diff
@@ -27,25 +27,25 @@ import "validate/validate.proto";
-  string                     originator_loan_id       = 3 [(validate.rules).string.min_len = 1];      // Originator's internal loan identifier
-  tech.figure.util.v1beta1.Borrowers borrowers                = 4 [(validate.rules).message.required = true]; // Borrower(s), co-signers, etc
-  Terms                      terms                    = 5 [(validate.rules).message.required = true]; // Loan terms: amount, duration, rates
-  Funding                    funding                  = 6 [(validate.rules).message.required = true]; // Funding detail
-  ApplicationData            app                      = 7;                                            // Data collected by LOS loan application process and used for underwriting
-  Underwriting               underwriting             = 8;                                            // Underwriting
-  repeated tech.figure.util.v1beta1.DocumentMetadata doc_meta = 9;                                            // Metadata about documents related to the loan (document files stored separately)
-  repeated tech.figure.util.v1beta1.SignedData signed_data    = 10;                                           // Digitally-signed-by-source data/documents
+  tech.figure.util.v1beta1.UUID                       id                 = 1 [(validate.rules).message.required = true]; // Loan UUID identifier
+  string                                              originator_name    = 2 [(validate.rules).string.min_len = 1];      // Human-readable name of the originating firm
+  string                                              originator_loan_id = 3 [(validate.rules).string.min_len = 1];      // Originator's internal loan identifier
+  tech.figure.util.v1beta1.Borrowers                  borrowers          = 4 [(validate.rules).message.required = true]; // Borrower(s), co-signers, etc
+Terms                                                 terms              = 5 [(validate.rules).message.required = true]; // Loan terms: amount, duration, rates
+  Funding                                             funding            = 6 [(validate.rules).message.required = true]; // Funding detail
+  ApplicationData                                     app                = 7;                                            // Data collected by LOS loan application process and used for underwriting
+  Underwriting                                        underwriting       = 8;                                            // Underwriting
+  repeated tech.figure.util.v1beta1.DocumentMetadata  doc_meta           = 9;                                            // Metadata about documents related to the loan (document files stored separately)
+  repeated tech.figure.util.v1beta1.SignedData        signed_data        = 10;                                           // Digitally-signed-by-source data/documents
   /*
   Loan-type specific information resides in one of the following loan types
    */
   oneof type_specific {
```

Comment:
Should we pull `asset_type` out of this `oneof`? 
In the `AssetType` definition we give example for `supertype` of MORTGAGE, HELOC, PERSONAL_LOAN, AUTO_LOAN. Since the `asset_type is within the `oneof` here, we couldn't both set `asset_type.supertype = MORTGAGE` and provide a `mortgage` proto.

Reply:
Not sure on what you mean by we couldn't do both. Are you suggesting to put it into the base loan message?

Reply:
the original intention of the oneof was that the type is implied by the proto (e.g. `Mortgage` proto) and that if you don't supply that proto, you are forced by the oneof requirements to at least say what your loan type is. (this wasn't my design idea, but i see the point.)

i'd be fine with pulling the type out of the oneof and letting it be top-level... it's easier to write code that switches on a type field than to be doing introspection into the oneof message type

Reply:
If we make any non-backwards compatible changes, we should instead start making the `v1` package version.

Reply:
Good points. I'll deprecate those fields